craft_application/models/project.py

```diff
@@ -114,12 +113,6 @@ def effective_base(self) -> Any:  # noqa: ANN401 app specific classes can improv
             return self.base
         raise RuntimeError("Could not determine effective base")
 
-    def get_build_plan(self) -> List[BuildInfo]:
```

Comment:
We should assess how much this is [going to break the tools](https://github.com/canonical/rockcraft/blob/17d6a0447f71b4f32657500e876903fa8be9f74c/rockcraft/services/package.py#L73)